### ISSUE TITLE
`libnp_helper.dylib` incompatible architecture on Mac M1 with Python 3.11

### ISSUE THREAD

**Issue by matteoacrossi** [pyscf/pyscf#1851]
I am installing `pyscf` 2.3.0 on a fresh python 3.11.4 environment (mambaforge) on a M1 Mac. I am using `pip install pyscf`, which then installs

```
h5py==3.9.0
numpy==1.25.2
pyscf==2.3.0
scipy==1.11.2
```

If I import `numpy_helper` I get the following error

```python
>>> from pyscf import numpy_helper
Traceback (most recent call last):
  File "/Users/matteo/mambaforge/envs/testpy311/lib/python3.11/site-packages/pyscf/lib/misc.py", line 90, in load_library
    return numpy.ctypeslib.load_library(libname, _loaderpath)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/matteo/mambaforge/envs/testpy311/lib/python3.11/site-packages/numpy/ctypeslib.py", line 156, in load_library
    return ctypes.cdll[libpath]
           ~~~~~~~~~~~^^^^^^^^^
  File "/Users/matteo/mambaforge/envs/testpy311/lib/python3.11/ctypes/__init__.py", line 451, in __getitem__
    return getattr(self, name)
           ^^^^^^^^^^^^^^^^^^^
  File "/Users/matteo/mambaforge/envs/testpy311/lib/python3.11/ctypes/__init__.py", line 446, in __getattr__
    dll = self._dlltype(name)
          ^^^^^^^^^^^^^^^^^^^
  File "/Users/matteo/mambaforge/envs/testpy311/lib/python3.11/ctypes/__init__.py", line 376, in __init__
    self._handle = _dlopen(self._name, mode)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^
OSError: dlopen(/Users/matteo/mambaforge/envs/testpy311/lib/python3.11/site-packages/pyscf/lib/libnp_helper.dylib, 0x0006): tried: '/Users/matteo/mambaforge/envs/testpy311/lib/python3.11/site-packages/pyscf/lib/libnp_helper.dylib' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64')), '/System/Volumes/Preboot/Cryptexes/OS/Users/matteo/mambaforge/envs/testpy311/lib/python3.11/site-packages/pyscf/lib/libnp_helper.dylib' (no such file), '/Users/matteo/mambaforge/envs/testpy311/lib/python3.11/site-packages/pyscf/lib/libnp_helper.dylib' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64'))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/matteo/mambaforge/envs/testpy311/lib/python3.11/site-packages/pyscf/__init__.py", line 97, in <module>
    from pyscf import lib
  File "/Users/matteo/mambaforge/envs/testpy311/lib/python3.11/site-packages/pyscf/lib/__init__.py", line 24, in <module>
    from pyscf.lib import numpy_helper
  File "/Users/matteo/mambaforge/envs/testpy311/lib/python3.11/site-packages/pyscf/lib/numpy_helper.py", line 39, in <module>
    _np_helper = misc.load_library('libnp_helper')
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/matteo/mambaforge/envs/testpy311/lib/python3.11/site-packages/pyscf/lib/misc.py", line 98, in load_library
    return numpy.ctypeslib.load_library(libname, libpath)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/matteo/mambaforge/envs/testpy311/lib/python3.11/site-packages/numpy/ctypeslib.py", line 156, in load_library
    return ctypes.cdll[libpath]
           ~~~~~~~~~~~^^^^^^^^^
  File "/Users/matteo/mambaforge/envs/testpy311/lib/python3.11/ctypes/__init__.py", line 451, in __getitem__
    return getattr(self, name)
           ^^^^^^^^^^^^^^^^^^^
  File "/Users/matteo/mambaforge/envs/testpy311/lib/python3.11/ctypes/__init__.py", line 446, in __getattr__
    dll = self._dlltype(name)
          ^^^^^^^^^^^^^^^^^^^
  File "/Users/matteo/mambaforge/envs/testpy311/lib/python3.11/ctypes/__init__.py", line 376, in __init__
    self._handle = _dlopen(self._name, mode)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^
OSError: dlopen(/Users/matteo/mambaforge/envs/testpy311/lib/python3.11/site-packages/pyscf/lib/libnp_helper.dylib, 0x0006): tried: '/Users/matteo/mambaforge/envs/testpy311/lib/python3.11/site-packages/pyscf/lib/libnp_helper.dylib' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64')), '/System/Volumes/Preboot/Cryptexes/OS/Users/matteo/mambaforge/envs/testpy311/lib/python3.11/site-packages/pyscf/lib/libnp_helper.dylib' (no such file), '/Users/matteo/mambaforge/envs/testpy311/lib/python3.11/site-packages/pyscf/lib/libnp_helper.dylib' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64'))
```

Everything works fine with Python 3.10

**Comment by jamesETsmith** [comment]
It looks like something is still wrong with the Python 3.11 wheels after #1791 and #1802. @sunqm or @PhilipVinc, do you have any ideas about this?

**Comment by PhilipVinc** [comment]
I had seen this after you merged the pr using cibuildwheel a few months back. Might be outdated nowadays but...

I think the universal (arm+x86) wheels you uploaded to pypi are broken and not actually universal, as per my previous issue. 

You never deleted those wheels.

And you never regenerated arm-only (or new universal) wheels with the new cibuildwheel infrastructure.

**Comment by matteoacrossi** [comment]
I suppose this is duplicate of #1791 then?

**Comment by PhilipVinc** [comment]
My impression is that that issue has been technically addressed but no wheels have been uploaded/amended with the new infrastructure

**Comment by mkrompiec** [comment]
It would be great if the updated wheels got uploaded soon, this is stopping us from moving our workflow to Python 3.11

**Comment by sunqm** [comment]
Wheels for Apple silicon have been available since Pyscf-2.5. If you encounter a similar issue in the new release, please feel free to reopen this issue.
